docs: camera-enablement

jwwojak opened the pull request:
# Overview

Updates the Flex manual with basic information about the built-in camera.

Sandbox: 

- Opentrons App: https://sandbox.docs.opentrons.com/docs-camera-enablement/flex/opentrons-app/
(see also a bullet in the "Recent Protocol Runs" section that follows. Explains downloading images.)

- Touchscreen > robot settings: https://sandbox.docs.opentrons.com/docs-camera-enablement/flex/touchscreen/settings/

- System Description > Frame and enclosure: https://sandbox.docs.opentrons.com/docs-camera-enablement/flex/system-description/robot/

RTC-884

## Test Plan and Hands on Testing

Making sure images and links work.

## Changelog

**Opentrons App chapter**: Modifies `opentrons-app.md` to include a new section about the camera. Additional changes (more camera features) will come with 8.9, so this first version is basic functionality.

**Touchscreen > Robot Settings**: Modifies `settings.md`. Updates the ODD image that shows all the settings. This is `touchscreen-settings.png`. No new text or text changes.

**System Description**: Modifies `robot.md`. In the "Frame and Enclosure" section, adds link to Opentrons App chapter to text in the sentence, "A 2.1-megapixel camera can photograph the deck and working area..."


## Review requests

The "Camera features and controls" section in the App chapter is the most significant change here. It introduces new information about the camera. This first camera release is limited, other features (e.g. shown in the Figma designs) aren't here and will be released later. Remember, this focuses on the very basic features and functions.

## Risk assessment

Low but important. Want to give users enough to use the camera easily.

jwwojak committed:
End of day save

jwwojak committed:
End of day save

jwwojak committed:
late day commit

I need to save before something bad happens

jwwojak committed:
late day commit

I need to save before something bad happens

jwwojak committed:
End of day save

Struggling with images.

jwwojak committed:
End of day save

Struggling with images.

jwwojak committed:
Morning save

jwwojak committed:
Morning save

jwwojak committed:
Main section content

jwwojak committed:
Main section content

jwwojak committed:
Update touchscreen-settings image

Includes camera enablement

jwwojak committed:
Update touchscreen-settings image

Includes camera enablement

jwwojak committed:
Revisions to other pages

- settings.md
- robot.md

jwwojak committed:
Revisions to other pages

- settings.md
- robot.md

jwwojak committed:
Header revisions

jwwojak committed:
Header revisions

jwwojak committed:
Better default camera image and angle

Removing this weird thing on line 166

jwwojak committed:
Better default camera image and angle

Removing this weird thing on line 166

jwwojak committed:
Reverting to older settings image

Shorter list, excludes camera prefs.

jwwojak committed:
Reverting to older settings image

Shorter list, excludes camera prefs.

ecormany commented on docs/flex/docs/system-description/robot.md:
```diff
@@ -13,7 +13,7 @@ The *frame* of the Opentrons Flex robot provides rigidity and structural support
 The metal frame has openings for *side windows* and a *front door* made of transparent polycarbonate that let you see what's going on inside Flex. The front door hinges open for access to the interior of the system. With the front door open, you can attach instruments, modules, and deck fixtures; prepare the deck before a protocol; or manipulate the state of the deck during a protocol.
 
-White LED strips on the inside top edges of the frame provide software-controllable ambient lighting. A 2-megapixel camera can photograph the deck and working area for recording and tracking protocol execution.
+White LED strips on the inside top edges of the frame provide software-controllable ambient lighting. A 2.1-megapixel camera can [photograph the deck](../opentrons-app.md#camera-features-and-controls) and working area for recording and tracking protocol execution.
```
We may have to leave this at 2, since 2.1 is rounding up full HD resolution (1920*1080=2,073,600).

jwwojak replied:
From my conversations about the camera with engineers, this is a 2.1 megapixel full HD camera. In fact Jamey, Casey, and Anthony found text like this acceptable. From Slack, https://opentrons.slack.com/archives/C08TTAGGLUX/p1763147541847549

> So, does this sound ok:
"The Flex uses a 2.1 megapixel camera to capture full HD images (1920x1080 pixels) of the deck."

In the interest of getting 'er done, this can go back to 2.

ecormany commented on docs/flex/docs/opentrons-app.md:
```diff
@@ -97,6 +97,48 @@ Click the three-dot menu (⋮) on the module card to choose from available comma
 
+## Camera features and controls
+
+Every Flex comes equipped with a built-in 2.1 megapixel camera that can capture full HD still images and provides video of the deck and working area. Disabled until the release of robot software version 8.8, the camera is now enabled for control and  use in the Opentrons App and from the touchscreen. When on, the camera provides:
```
I think this is too negative, even if this has been a historical gap.
```suggestion
Every Flex comes equipped with a built-in 2.1 megapixel camera that can capture full HD still images and provides video of the deck and working area. Starting in robot software version 8.8, you can control the camera in the Opentrons App and from the touchscreen. When enabled, the camera provides:
```

jwwojak replied:
Thank you. Applying suggested change.

ecormany commented on docs/flex/docs/opentrons-app.md:
```diff
@@ -97,6 +97,48 @@ Click the three-dot menu (⋮) on the module card to choose from available comma
+
+This fixed location gives you a wide-angle field of view of attached instruments, deck-mounted modules, and labware.
+
+![Default camera view](./images/camera-default.png) 
```
This section seems better suited to the system description section. Let's move it there to keep this page focused on app features.

jwwojak replied:
Sure, will move it!

ecormany commented on docs/flex/docs/opentrons-app.md:
```diff
@@ -97,6 +97,48 @@ Click the three-dot menu (⋮) on the module card to choose from available comma
+
+</div>
+
+You can also see the camera's status, and turn it on and off, from the Peripherals section of the details page for your Flex. Click the three-dot menu ( ⋮ ) to enable or disable the camera. This example shows a Flex with its camera enabled.
```
We now have a mix of `( ⋮ )` and `(⋮)` in the manual. We should pick which one we think looks better and standardize (in a separate PR).

jwwojak replied:
Revising to use (⋮)

ecormany commented on docs/flex/docs/opentrons-app.md:
```diff
@@ -114,4 +156,6 @@ Each entry in the recent protocol runs list includes the protocol name, its time
 !!! note
     If you need to maintain a comprehensive record of all runs performed on your Flex, you must use the **Download run log** feature to save this information to your computer.
```
It's strange that this note breaks up the list here. Either:
- Put it after the entire list (most like previous revision).
- Put it inside the `Download run log` bullet (indented so it's contained in that list item).

jwwojak replied:
This is maybe for the other PRs that got merged together, not for the camera PR?

ecormany commented on docs/flex/docs/opentrons-app.md:
```diff
@@ -114,4 +156,6 @@ Each entry in the recent protocol runs list includes the protocol name, its time
 !!! note
     If you need to maintain a comprehensive record of all runs performed on your Flex, you must use the **Download run log** feature to save this information to your computer.
 
+- **Download image files:** Save to your computer a ZIP file containing all the still images taken during a protocol run, if the camera was enabled.
```
Apparently [nobody can decide](https://books.google.com/ngrams/graph?content=zip+file&year_start=1990&year_end=2022&corpus=en&smoothing=3&case_insensitive=true) whether it's "ZIP" or "zip". Even [the Wikipedia article](https://en.wikipedia.org/wiki/ZIP_(file_format)) mixes usage. We have cleverly avoided the problem in the rest of the manual by using the file extension, and we can do that again here.

```suggestion
- **Download image files:** Save to your computer a `.zip` file containing all the still images taken during a protocol run, if the camera was enabled.
```

jwwojak replied:
Applying suggestion.

ecormany reviewed (commented):
Thanks for updating this draft to avoid the issues with the settings screen; we'll pick those up later.

Content is good, but I think it could be reshuffled a bit to best fit in with the existing structure of the two modified pages.

mjhuff reviewed (commented):
Just commenting that all of this is accurate from a SW POV. Thank you!

jwwojak committed:
Merge branch 'docs-camera-enablement' of https://github.com/Opentrons/opentrons into docs-camera-enablement

jwwojak committed:
Apply suggestions from code review

Co-authored-by: Ed Cormany <edward.cormany@opentrons.com>

jwwojak committed:
Changes from reviewers

- Change camera specs to "2-megapixel"
- Revert to original ODD image in Robot Settings
- Move some camera details text to system description.
